internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -451,6 +451,28 @@ func flattenProcessArgs(p *admin20231115.ClusterDescriptionProcessArgs) []map[st
 
 func FlattenAdvancedReplicationSpecsOldSDK(ctx context.Context, apiObjects []admin20231115.ReplicationSpec, rootDiskSizeGB float64, tfMapObjects []any,
 	d *schema.ResourceData, connV2 *admin.APIClient) ([]map[string]any, error) {
+	replicationSpecFlattener := func(ctx context.Context, sdkModel *admin20231115.ReplicationSpec, tfModel map[string]any, resourceData *schema.ResourceData, client *admin.APIClient) (map[string]any, error) {
+		return flattenAdvancedReplicationSpecOldSDK(ctx, sdkModel, rootDiskSizeGB, tfModel, resourceData, connV2)
+	}
+	return flattenAdvancedReplicationSpecsLogic[admin20231115.ReplicationSpec](ctx, apiObjects, tfMapObjects, d,
+		doesAdvancedReplicationSpecMatchAPIOldSDK, replicationSpecFlattener, connV2)
+}
+
+func flattenAdvancedReplicationSpecs(ctx context.Context, apiObjects []admin.ReplicationSpec20240710, tfMapObjects []any,
```

Comment:
same here: `flattenAdvancedReplicationSpecs` and `flattenAdvancedReplicationSpecsOldSDK` suite very well for some sort of design pattern and I believe Strategy is a good one but there might be others

Reply:
In the case of `flattenAdvancedReplicationSpecs` and `FlattenAdvancedReplicationSpecsOldSDK` both share the same logic, just different model and function for matching tf model with API model. This is why I went with the approach of a common function with generics so we don't need to duplicate any of that logic.